File: Chuong 11 - Ke Thua/Ung Dung Quan Ly Thu Vien C++ - C# - Java/Source Code/Java/Java_QuanLyDocGia/src/java_quanlydocgia/DocGiaNguoiLon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package java_quanlydocgia;

/**
 *
 * @author NguyenVietNamSon
 */
public class DocGiaNguoiLon extends DocGia{
        private String CMND;

        DocGia x = new DocGia();
        public void Nhap()
        {
            x.Nhap(); // Gọi lại phương thức nhập của cha

            // CMND sẽ nhập trong thư viện
        }

        public void Xuat()
        {
            x.Xuat(); // Gọi lại phương thức xuất của cha

            System.out.print("\nCMND: " + getCMND());
            System.out.print("\nTien lam the: " + String.valueOf(TinhTienLamThe()));
        }

        public float TinhTienLamThe()
        {
            return x.SoThangHieuLuc * 10000;
        }

    /**
     * @return the CMND
     */
    public String getCMND() {
        return CMND;
    }

    /**
     * @param CMND the CMND to set
     */
    public void setCMND(String CMND) {
        this.CMND = CMND;
    }
}

File: Chuong 11 - Ke Thua/Ung Dung Quan Ly Thu Vien C++ - C# - Java/Source Code/C#/Quan Ly Doc Gia/QuanLyDocGia/DocGiaNguoiLon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyDocGia
{
    class DocGiaNguoiLon : DocGia
    {
        private string CMND;

        public string _CMND
        {
            get { return CMND; }
            set { CMND = value; }
        }

        public void Nhap()
        {
            base.Nhap(); // Gọi lại phương thức nhập của cha

            // CMND sẽ nhập trong thư viện
        }

        public void Xuat()
        {
            base.Xuat(); // Gọi lại phương thức xuất của cha

            Console.Write("\nCMND: {0}", CMND);
        }

        public float TinhTienLamThe()
        {
            return SoThangHieuLuc * 10000;
        }
    }
}
